pkg/client/cache/eventqueue.go

```diff
@@ -36,12 +36,13 @@ import (
 // watch.Deleted event (though this state is delivered by the watch API itself, it is not passed on
 // to the reflector Store).
 type EventQueue struct {
-	lock   sync.RWMutex
-	cond   sync.Cond
-	store  kcache.Store
-	keyFn  kcache.KeyFunc
-	events map[string]watch.EventType
-	queue  []string
+	lock           sync.RWMutex
+	cond           sync.Cond
+	store          kcache.Store
+	keyFn          kcache.KeyFunc
+	events         map[string]watch.EventType
+	queue          []string
+	lastReplaceKey string
```

Comment:
You need to add comments around this intent and why it works.  It took a very long time for me to understand that this works because a reflector will call `Replace` on a re-list and the compression algorithm doesn't shift locations.  That means that you got to the end of the initial list, plus any compressions made along the way.